hypothesis-python/src/hypothesis/ghostwriter.py

```diff
@@ -104,23 +146,53 @@ def _get_strategies(*funcs: Callable) -> Dict[str, st.SearchStrategy]:
             raise NotImplementedError("Expected to pass everything as kwargs")
 
         for k, v in zip(kwargs.keys, kwargs.mapped_strategy.element_strategies):
-            given_strategies[k] = v
+            if repr(given_strategies.get(k, v)) != repr(v):
+                given_strategies[k] = given_strategies[k] | v
+            else:
+                given_strategies[k] = v
+
+    if len(funcs) == 1:
+        indexer = list(_get_params(f)).index
+        return OrderedDict(
+            sorted(given_strategies.items(), key=lambda kv: indexer(kv[0]))
+        )
+    return given_strategies
+
+
+def _assert_eq(style, a, b):
+    return {
+        "pytest": "assert {0} == {1}, ({0}, {1})",
+        "unittest": "self.assertEqual({}, {})",
+    }[style].format(a, b)
 
-    indexer = list(_get_params(f)).index
-    return OrderedDict(sorted(given_strategies.items(), key=lambda kv: indexer(kv[0])))
+
+def _valid_syntax_repr(strategy):
+    # Return a syntactically-valid strategy repr, including fixing some
+    # strategy reprs and replacing invalid syntax reprs with `"nothing()"`.
+    r = (
+        repr(strategy)
+        .replace("sampled_from([False, True])", "booleans()")
+        .replace("just(None)", "none()")
```

Comment:
Some of these I wonder if we should replace in the strategy reprs themselves. e.g. it seems not unreasonable to have these special cases for `just` and `sampled_from` in their `__repr__` definition.